esmvaltool/diag_scripts/zmnam/zmnam.py

```diff
@@ -0,0 +1,99 @@
+"""
+;;#############################################################################
+;; Zonal mean Northern Annular Mode Diagnostics
+;; Author: Federico Serva (ISAC-CNR & ISMAR-CNR, Italy)
+;; Copernicus C3S 34a lot 2 (MAGIC)
+;;#############################################################################
+;; Description
+;;    Evaluation of stratosphere-troposphere coupling
+;;    based on EOF/PC analysis of the geopotential height field
+;; Modification history
+;;    20180512-A_serv_fe: Added output netCDFs, more use of preprocessor
+;;    20180510-A_serv_fe: Routines written.
+;;#############################################################################
+"""
+
+import yaml
+import sys
+import cartopy.crs as ccrs
+import matplotlib.pyplot as plt
+import warnings
+import iris
+import iris.plot as iplt
+import iris.quickplot as qplt
+import os
+import logging
+
+from esmvaltool.diag_scripts.shared import (group_metadata, run_diagnostic,
+                                            select_metadata, sorted_metadata)
+from esmvaltool.diag_scripts.shared.plot import quickplot
+
+logger = logging.getLogger(__name__)
+
+if not sys.warnoptions:
+    warnings.simplefilter("ignore")
+
+
+def get_input_files(cfg, index=0):
+    """Get a dictionary with input files from metadata.yml files."""
+    metadata_file = cfg['input_files'][index]
+    with open(metadata_file) as file:
+        metadata = yaml.safe_load(file)
+    return metadata
+
+
+def main(cfg):
+
+    logger.setLevel(cfg['log_level'].upper())
+
+    input_files = get_input_files(cfg)
+
+    plot_dir = cfg['plot_dir']
+    out_dir = cfg['work_dir']
+    write_plots = cfg['write_plots']
+    fig_fmt = cfg['output_file_type']
+
+    # Import full diagnostic routines
+    from zmnam_calc import zmnam_calc
```

Comment:
Please move your imports to the top of the module

Reply:
Yes, this is also suggested by the Codacy test. I'll try to fix as much as possible

Reply:
A number of codacy issues are now solved